rwmutex.go

```diff
@@ -113,8 +124,20 @@ func (m *RWMutex) TryRLock() bool {
 // It is a run-time error if rw is not locked for reading
 // on entry to RUnlock.
 func (m *RWMutex) RUnlock() {
-	m.wg.Done()
-	m.mutex.RUnlock()
+	defer m.mutex.RUnlock()
+
+	if DebugIsOn {
+		m.wg.Done()
+		defer func() {
+			err := recover()
+			if err != nil {
```

Comment:
Could we use `errors.Is` to verify that the err is indeed because we went below 0, or match a string ? 
( I don't remember if `recover` provides an error or a string, and if it's a string if we have a type to compare it to )

I'm just thinking it would be a pity to "hide" an unrelated panic, somehow.

Reply:
"The return value of recover is the error raised in the call to panic."  Hard to use a `Is(...)` in this case. But maybe I could check if the wg goes bellow zero ?